lib/utils/sorted_list.dart

```diff
@@ -0,0 +1,93 @@
+abstract class AuList<T> {
+  late List<T> _list;
+
+  List<T> get items => _list;
+
+  int get length => _list.length;
+
+  bool get isNotEmpty => _list.isNotEmpty;
+
+  bool get isEmpty => _list.isEmpty;
+
+  T get first => _list.first;
```

Comment:
after changing the `items` to the `unmodifiable` list then I suggest using `items.first` and `items.last` for this